From 17d2509ebf7bcf7fbfb6b55747365c79e77ca99c Mon Sep 17 00:00:00 2001
From: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Date: Fri, 10 Sep 2021 13:55:00 +0300
Subject: [PATCH 5/5] s32r45: SoC: Print SoC revision Major.Minor

As part of S32R45 SoC rev. 2.0 MIDR rev ID register major
mask was updated to value 0001.
We print this information in u-boot.

Upstream-Status: Pending

Issue: ALB-6111
Signed-off-by: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/s32-gen1/soc.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/soc.c b/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
index 84256bea54..0d4ad571c8 100644
--- a/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
+++ b/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
@@ -169,7 +169,10 @@ int print_cpuinfo(void)
 		   get_siul2_midr2_subminor());
 	#endif  /* CONFIG_TARGET_TYPE_S32GEN1_SIMULATOR */
 #elif defined(CONFIG_NXP_S32R45)
-	printf("CPU:\tNXP S32R45\n");
+	printf("CPU:\tNXP S32R45");
+	printf(" rev. %d.%d\n",
+	       get_siul2_midr1_major() + 1,
+	       get_siul2_midr1_minor());
 #elif defined(CONFIG_NXP_S32G3XX)
 	printf("CPU:\tNXP S32G398A\n");
 #endif
-- 
2.17.1

